### DISCUSSION TITLE
JSL: how to include .css also?

### DISCUSSION THREAD

**Discussion by seek78** [dnnsoftware/Dnn.Platform#5721] [Q&A]
Hi,
looking into the JSL implementation, I found JQueryUI is included "simply" by calling: 

> `RequestRegistration(CommonJs.jQueryUI)`

which call

> `RequestRegistration(jsname, null, SpecificVersion.Latest)`

My goal is to add some JSL in order to help developers, but I need to include also their StyleSheet file like JQueryUI.
I cannot understand where the .css of JQueryUI where included, how "think" DNN?

Thanks.
S.

**Comment by valadas** [comment]
What is JSL ?

**Reply by seek78** [reply]
JSL = JavaScript Libraries.
I found the acronym in DNN sources

**Comment by bdukes** [comment]
DNN does this for jQuery UI by including the CSS for the dialog in `default.css`, not by any way of magically including the CSS when the JS is requested. There have been suggestions to make this easier, e.g. #3382, but no contributions to make it work.

When we've needed to do this, we include the CSS file with the JS library, and then just hardcode the path to the CSS, including the version number for the JS library, to include it directly. It's pretty fragile.
